lib/src/css_parser.dart

```diff
@@ -809,34 +809,58 @@ class ExpressionMapping {
+      css.Expression? offsetX;
+      css.Expression? offsetY;
+      css.Expression? blurRadius;
+      css.HexColorTerm? color;
+      int expressionIndex = 0;
+      list.forEach((element) {
+        if (element is css.HexColorTerm) {
+          color = element;
+        } else if (expressionIndex == 0) {
+          offsetX = element;
+          expressionIndex++;
+        } else if (expressionIndex++ == 1) {
+          offsetY = element;
+          expressionIndex++;
+        } else {
+          blurRadius = element;
+        }
+      });
       RegExp nonNumberRegex = RegExp(r'\s+(\d+\.\d+)\s+');
-      if (exp is css.LiteralTerm && exp2 is css.LiteralTerm) {
-        if (exp3 != null && ExpressionMapping.expressionToColor(exp3) != null) {
-          shadow.add(Shadow(
-              color: expressionToColor(exp3)!,
-              offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, ''))!, double.tryParse(exp2.text.replaceAll(nonNumberRegex, ''))!)
-          ));
-        } else if (exp3 != null && exp3 is css.LiteralTerm) {
-          if (exp4 != null && ExpressionMapping.expressionToColor(exp4) != null) {
+      if (offsetX is css.LiteralTerm && offsetY is css.LiteralTerm) {
+        if (color != null && ExpressionMapping.expressionToColor(color) != null) {
+          if (blurRadius is css.LiteralTerm) {
             shadow.add(Shadow(
-                color: expressionToColor(exp4)!,
-                offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, ''))!, double.tryParse(exp2.text.replaceAll(nonNumberRegex, ''))!),
-                blurRadius: double.tryParse(exp3.text.replaceAll(nonNumberRegex, ''))!
+                color: expressionToColor(color)!,
+                offset: Offset(
+                    double.tryParse((offsetX as css.LiteralTerm).text.replaceAll(nonNumberRegex, ''))!,
+                    double.tryParse((offsetY as css.LiteralTerm).text.replaceAll(nonNumberRegex, ''))!),
+                blurRadius: double.tryParse((blurRadius as css.LiteralTerm).text.replaceAll(nonNumberRegex, ''))!
             ));
           } else {
             shadow.add(Shadow(
-                offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, ''))!, double.tryParse(exp2.text.replaceAll(nonNumberRegex, ''))!),
-                blurRadius: double.tryParse(exp3.text.replaceAll(nonNumberRegex, ''))!
+                color: expressionToColor(color)!,
+                offset: Offset(
+                    double.tryParse((offsetX as css.LiteralTerm).text.replaceAll(nonNumberRegex, ''))!,
+                    double.tryParse((offsetY as css.LiteralTerm).text.replaceAll(nonNumberRegex, ''))!),
             ));
           }
         } else {
-          shadow.add(Shadow(
-              offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, ''))!, double.tryParse(exp2.text.replaceAll(nonNumberRegex, ''))!)
-          ));
+          if (blurRadius is css.LiteralTerm) {
```

Comment:
The differnece is only that added `blurRadius` parameter? If so, can;t you do something like 
```
blurRadius: (blurRadius is css.LiteralTerm) ? double.tryParse((blurRadius as css.LiteralTerm).text.replaceAll(nonNumberRegex, ''))! : null,
```
this would make much less code and by extension make it easier to understand